### PR TITLE
feat: support dns-result-order Node.js cli flag

### PR DESCRIPTION
**Pull request by deepak1556**
#### Description of Change

This allows applications that load 3-party plugins to provide the [legacy behavior](https://github.com/nodejs/node/commit/1b2749ecbe) with a process wide flag. There is also [dns.setDefaultResultOder](https://nodejs.org/dist/latest-v18.x/docs/api/dns.html#dnssetdefaultresultorderorder) but it is per thread API so cannot be used effectively in a plugin process.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Add support for --dns-result-order Node.js cli flag

### CODE REVIEW

File: docs/api/command-line-switches.md [hunk 1]
```diff
@@ -291,6 +291,15 @@ Print stack traces for deprecations.
...
 
+### `--dns-result-order=order`
+
+Set the default value of verbatim in `dns.lookup()` and `dnsPromises.lookup()`. The value could be:
+
+* ipv4first: sets default verbatim false.
+* verbatim: sets default verbatim true.
+
+The default is verbatim and `dns.setDefaultResultOrder()` have higher priority than `--dns-result-order`.
+
```
Comment (erickzhao) [inline docs/api/command-line-switches.md:302]: Added the links and formatting back from the Node.js docs and changed the wording a bit for clarity:

```suggestion
### `--dns-result-order=order`

Set the default value of the `verbatim` parameter in the Node.js [`dns.lookup()`](https://nodejs.org/api/dns.html#dnslookuphostname-options-callback) and [`dnsPromises.lookup()`](https://nodejs.org/api/dns.html#dnspromiseslookuphostname-options) functions.

The value could be:

* `ipv4first`: sets default `verbatim` `false`.
* `verbatim`: sets default `verbatim` `true`.

The default is `verbatim` and `dns.setDefaultResultOrder()` have higher priority than `--dns-result-order`.

```

**Review by codebytere — APPROVED** [review]
API LGTM

**Review by zcbenz — APPROVED** [review]
API LGTM

**Comment by zcbenz** [comment]
The patch fails to apply.

**Comment by deepak1556** [comment]
`main` is now on `v18.17.0` which contains `dns.getDefaultResultOrder`. Removed the patch.

**Pull request merged**